walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +16,65 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+
+    private final int endOfDay = TimeRange.END_OF_DAY;
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        // Collects data from the request: meeting duration and the attendees and optional attendees
+        int duration = (int) request.getDuration();
+        Collection<String> attendees = request.getAttendees();
+        Collection<String> opAttendees = request.getOptionalAttendees();
+
+
+        // This is the collection of time ranges that we will return that fits everyones schedule
+        Collection<TimeRange> unavailableTimes = new ArrayList<>();
+        Collection<TimeRange> availableTimes = new ArrayList<>();
+
+        int startTime = 0;
+        int endTime = startTime + duration;
+
+
+        // Finds all unavailable times
+        while( startTime < endOfDay ) {
+            TimeRange unavailableMeetTime = TimeRange.fromStartDuration(startTime, duration);
+            
+            for (Event event : events) {
+                TimeRange eventMeetTime = event.getWhen();
+                Collection<String> eventAttendees = event.getAttendees();
+
+                if (eventMeetTime.overlaps(unavailableMeetTime)) {
+                    if(!Collections.disjoint(eventAttendees, attendees)) {
+                        unavailableTimes.add(unavailableMeetTime);
+                    }
+                } 
+
+            }
+            startTime = startTime + duration;
```

Comment:
I think there are some issues with this solution, but I think the point of the exercise is to be able to catch them with tests, so I'll defer my comments for later.